src/client.c

```diff
@@ -221,9 +221,22 @@ static bool login_via_cert(PgSocket *client)
 		goto fail;
 
 	log_debug("TLS cert login: %s", tls_peer_cert_subject(client->sbuf.tls));
+
 	if (!tls_peer_cert_contains_name(client->sbuf.tls, client->login_user->name)) {
-		slog_error(client, "TLS certificate name mismatch");
-		goto fail;
+		log_debug("TLS certificate name mismatch. Looking for a map.");
+		if (!rule)
+			goto fail;
+
+		if (rule->identmap == NULL)
+			goto fail;
+
+		if (!tls_peer_cert_contains_name(client->sbuf.tls, rule->identmap->system_user_name))
+			goto fail;
+
+		if (strcmp(client->login_user->name, rule->identmap->postgres_user_name)) {
+			if (strcmp(rule->identmap->postgres_user_name, "all"))
```

Comment:
There's two ways all can be formatted in the ident file, which means two different things:
- `all` (literal string), this is the special keyword that matches all users
- `"all"` (quoted), this means it only matches the user called `all` (assuming this user exists)

Postgres tracks this difference by setting a flag in case the special keyword is used.

Reply:
For HBA parsing we use the `NAME_ALL` flag in pgbouncer too actually.